src/utils/rate-limit.js

```diff
@@ -25,27 +25,30 @@
  * @returns {function()}
  */
 export function rateLimit(win, callback, minInterval) {
-  let locker = null;
+  let locker = 0;
   let nextCallArgs = null;
 
-  const fire = args => {
-    callback.apply(null, args);
-
+  function fire(args) {
     nextCallArgs = null;
     // Lock the fire for minInterval milliseconds
-    locker = win.setTimeout(() => {
-      locker = null;
-      // If during the period there're invocations queued up, fire once.
-      if (nextCallArgs !== null) {
-        fire(nextCallArgs);
-      }
-    }, minInterval);
-  };
+    locker = win.setTimeout(waiter, minInterval);
+
+    callback.apply(null, args);
+  }
+
+  function waiter() {
```

Comment:
Every time you called the old `fire`, it would create a brand new instance of the arrow function. Hoisting it out means it's instantiated and reused between `fire` calls.